odl/discr/grid.py

```diff
@@ -54,6 +54,7 @@ def sparse_meshgrid(*x):
         xi = np.asarray(xi)
         slc = [None] * n
         slc[ax] = slice(None)
+        slc = tuple(slc)
```

Comment:
Code like this without conversion from list to tuple raises a `DeprecationWarning` in NumPy 1.15, we should apply the fix throughout.